src/std/math.earl

```diff
@@ -42,3 +44,102 @@ module Math
     }
     return x;
 }
+
+# Statistical Functions
+
+# Mode
+# This function returns a list of all modes that appear at the
+# highest frequency in the given list in the order modes are found.
+@pub fn mode(lst) {
+    
+    if len(lst) == 0 {
+        return none;
+    }
+
+    # if len(lst) == 1 {
+    #     return lst[0];
+    # }
+    
+    let count = 0;
+    let max_count = 0;
+    let lst_modes = [];
+    
+    for i in 0 to len(lst) {
+        if (List::find(lst_modes, lst[i]) == -1) {
+
+            count = List::count(lst, lst[i]);
+            
+            if max_count == count {
+                lst_modes.append(lst[i]);
+                count = 0;
+            }
+
+            if max_count < count {
+                lst_modes = [];
+                lst_modes.append(lst[i]);
+                max_count = count;
+                count = 0;
+             }
+        }
+    }
+    return lst_modes;
+}
+
+# Mean
+# This function returns the mean (Average in a dataset) of a given list
+@pub fn mean(lst) {
+    return List::sumf(lst) / len(lst);
+}
+
+# Median
+# This function sorts and then returns the middle number of a given list
+@pub fn median(lst) {
+    
+    List::quicksort(lst, List::DEFAULT_INT_ASCEND_QUICKSORT);
+    
+    let middle = len(lst)/2;
+    
+    if len(lst) % 2 == 0 {
+        return (lst[middle-1]+lst[middle])/2.0;
+    }
+
+    return lst[middle];
+}
+
+# min
+@pub fn min(lst) {
+
+    let x = 0;
+    let min = 0;
+    for i in len(lst) {
+        x = lst[i];
+        if min >= x {
+            min = x;
+        }
+    }
+
+    return min;
+}
+
+# max
+@pub fn max(lst) {
+    let x = 0;
+    let max = 0;
+    for i in len(lst) {
+        x = lst[i];
+        if max <= x {
+            max = x;
+        }
+    }
+    return max;
+}
+
+
+
+    
+# Mathematical Constants
+# pi
+@pub let pi = 3.141592653589793;
+
+# Euler's number 
+@pub let e = 2.718281828459045;
```

Comment:
give this a better name instead of just `e` as it could conflict with something else on accident if any future functions are marked as `@world`.

Reply:
Math::e should be fine but what were you thinking